api/api-internal-scms-v1/src/main/java/com/thoughtworks/go/apiv1/internalscms/InternalSCMsControllerV1.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.apiv1.internalscms;
+
+import com.thoughtworks.go.api.ApiController;
+import com.thoughtworks.go.api.ApiVersion;
+import com.thoughtworks.go.api.CrudController;
+import com.thoughtworks.go.api.base.OutputWriter;
+import com.thoughtworks.go.api.representers.JsonReader;
+import com.thoughtworks.go.api.spring.ApiAuthenticationHelper;
+import com.thoughtworks.go.api.util.GsonTransformer;
+import com.thoughtworks.go.apiv1.internalscms.representers.SCMRepresenter;
+import com.thoughtworks.go.apiv1.internalscms.representers.VerifyConnectionResultRepresenter;
+import com.thoughtworks.go.config.exceptions.EntityType;
+import com.thoughtworks.go.domain.scm.SCM;
+import com.thoughtworks.go.plugin.api.response.Result;
+import com.thoughtworks.go.server.service.EntityHashingService;
+import com.thoughtworks.go.server.service.materials.PluggableScmService;
+import com.thoughtworks.go.spark.Routes;
+import com.thoughtworks.go.spark.spring.SparkSpringController;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.function.Consumer;
+
+import static spark.Spark.*;
+
+@Component
+public class InternalSCMsControllerV1 extends ApiController implements SparkSpringController, CrudController<SCM> {
```

Comment:
We are not using this controller for CRUD operations. Does it still make sense to use `CrudController` interface?

Reply:
Yeah, you are right. It's not required, changed it.